src/main/java/wolox/training/models/User.java

```diff
@@ -53,13 +53,18 @@ public void setBooks(Collection<Book> books) {
         this.books = books;
     }
 
-    public Collection<Book> addBook(Book newBook) {
-        this.books.add(newBook);
-        return this.books;
+    public boolean addBook(Book newBook) {
+        if(this.books.stream().anyMatch(book -> book.getId() == newBook.getId())){
+            return false;
+        } else {
+            return this.books.add(newBook);
+        }
     }
 
     public Collection<Book> deleteBook(long bookToDeleteId){
-        this.books.removeIf(book -> book.getId() == bookToDeleteId);
+        if(this.books.stream().anyMatch(book -> book.getId() == bookToDeleteId)){
```

Comment:
If you check the docs of removeIf, you can see that it returns true if it deleted something, and false if not. Therefore, you can replac all of this for a simple "return this.books.removeIf(book -> book.getId() == bookToDeleteId);"